generic-extras/src/test/scala-2/io/circe/generic/extras/ConfiguredSemiautoDerivedSuite.scala

```diff
@@ -77,31 +76,6 @@ object ConfiguredSemiautoDerivedSuite {
 class ConfiguredSemiautoDerivedSuite extends CirceSuite {
   import ConfiguredSemiautoDerivedSuite._
 
-  checkAll("Codec[ConfigExampleBase]", CodecTests[ConfigExampleBase].codec)
-  checkAll(
-    "Codec[ConfigExampleBase] via Codec",
-    CodecTests[ConfigExampleBase](codecForConfigExampleBase, codecForConfigExampleBase).codec
-  )
-  checkAll(
-    "Codec[ConfigExampleBase] via Decoder and Codec",
-    CodecTests[ConfigExampleBase](implicitly, codecForConfigExampleBase).codec
-  )
-  checkAll(
-    "Codec[ConfigExampleBase] via Encoder and Codec",
-    CodecTests[ConfigExampleBase](codecForConfigExampleBase, implicitly).codec
-  )
-
-  property("Semi-automatic derivation should support configuration") {
-    forAll { (f: String, b: Double) =>
-      val foo: ConfigExampleBase = ConfigExampleFoo(f, 0, b)
-      val json = json"""{ "type": "config_example_foo", "this_is_a_field": $f, "b": $b}"""
-      val expected = json"""{ "type": "config_example_foo", "this_is_a_field": $f, "a": 0, "b": $b}"""
-
-      assert(Encoder[ConfigExampleBase].apply(foo) === expected)
-      assert(Decoder[ConfigExampleBase].decodeJson(json) === Right(foo))
-    }
-  }
-
   test("it should call field modification times equal to field count") {
```

Comment:
It fails in Scala 3, but this is implementation specific.

Looks like the `io.circe.derivation` implementation from `circe-core` calls the field modification functions on each `encode` / `decode`. 

It should be fixed in the `circe-core`.